src/nbl/asset/interchange/CGLTFLoader.h

```diff
@@ -0,0 +1,539 @@
+						{
+							std::optional<uint32_t> indices;									//!< The index of the accessor that contains the indices.
+							std::optional<uint32_t> material;									//!< The index of the material to apply to this primitive when rendering.
+							std::optional<uint32_t> mode;										//!< The type of primitives to render.
+							std::optional<std::unordered_map<std::string, uint32_t>> targets;	//!< An array of Morph Targets, each Morph Target is a dictionary mapping attributes (only POSITION, NORMAL, and TANGENT supported) to their deviations in the Morph Target.
+
+							/*
+								An accessor provides a typed view into a bufferView or a subset of a bufferView similar to
+								how WebGL's vertexAttribPointer() defines an attribute in a buffer.
+							*/
+
+							struct SGLTFAccessor
+							{
+								std::optional<uint32_t> bufferView;
+								std::optional<uint32_t> byteOffset;
+								std::optional<uint32_t> componentType;
+								std::optional<bool> normalized;
+								std::optional<uint32_t> count;
+								std::optional<std::string> type;
+								std::optional<std::vector<double>> max; // todo - common number types
+								std::optional<std::vector<double>> min; // todo - common number types
+								// TODO - sparse;
+								std::optional<std::string> name;
+
+								enum SCompomentType
+								{
+									SCT_BYTE = 5120,
+									SCT_UNSIGNED_BYTE = 5121,
+									SCT_SHORT = 5122,
+									SCT_UNSIGNED_SHORT = 5123,
+									SCT_UNSIGNED_INT = 5125,
+									SCT_FLOAT = 5126
+								};
+
+								struct SType
+								{
+									_NBL_STATIC_INLINE_CONSTEXPR std::string_view SCALAR = "SCALAR";
+									_NBL_STATIC_INLINE_CONSTEXPR std::string_view VEC2 = "VEC2";
+									_NBL_STATIC_INLINE_CONSTEXPR std::string_view VEC3 = "VEC3";
+									_NBL_STATIC_INLINE_CONSTEXPR std::string_view VEC4 = "VEC4";
+									_NBL_STATIC_INLINE_CONSTEXPR std::string_view MAT2 = "MAT2";
+									_NBL_STATIC_INLINE_CONSTEXPR std::string_view MAT3 = "MAT3";
+									_NBL_STATIC_INLINE_CONSTEXPR std::string_view MAT4 = "MAT4";
+								};
+
+								bool validate()
+								{
+									if (!componentType.has_value())
+										return false;
+
+									if (!count.has_value())
+										return false;
+									else
+										if (count.has_value() < 1)
+											return false;
+
+									if (!type.has_value())
+										return false;
+
+									return true;
+								}
+							};
+
+							/*
+								Valid attribute semantic property names include POSITION, NORMAL, TANGENT, TEXCOORD_0, TEXCOORD_1, COLOR_0, JOINTS_0, and WEIGHTS_0.
+								Application-specific semantics must start with an underscore, e.g., _TEMPERATURE. TEXCOORD, COLOR, JOINTS, and WEIGHTS attribute
+								semantic property names must be of the form [semantic]_[set_index], e.g., TEXCOORD_0, TEXCOORD_1, COLOR_0.
+							*/
+
+							std::unordered_map<std::string, SGLTFAccessor> accessors; //! An accessor is queried with an atribute name (like POSITION)
```

Comment:
why string and not an ENUM?

Reply:
what about accessors that support ID format like color attribute?

Reply:
explain on discord

Reply:
corrected in following commit